lib/prima_auth0_ex/telemetry.ex

```diff
@@ -0,0 +1,42 @@
+if Code.ensure_loaded?(Statix) do
+  defmodule PrimaAuth0Ex.Telemetry.Statix do
+    @moduledoc """
+    A pre-defined Statix module
+    """
+
+    alias PrimaAuth0Ex.Telemetry.Handler
+
+    use Statix, runtime_config: true
+
+    def setup do
+      :ok =
+        :telemetry.attach_many(
+          "auth0-handler",
+          [
+            [:prima_auth0_ex, :retrieve_token, :failure],
+            [:prima_auth0_ex, :retrieve_token, :success]
+          ],
+          &Handler.handle_event/4,
+          nil
+        )
+
+      :ok = __MODULE__.connect()
+    end
+  end
+
+  defmodule PrimaAuth0Ex.Telemetry.Handler do
+    @moduledoc """
+    A pre-defined Statix telemetry handler
+    """
+
+    alias PrimaAuth0Ex.Telemetry.Statix
+
+    def handle_event([:prima_auth0_ex, :retrieve_token, :failure], %{count: count}, %{audience: audience}, _config) do
+      Statix.increment("retrieve_token:failure", count, tags: ["audience:#{audience}"])
+    end
+
+    def handle_event([:prima_auth0_ex, :retrieve_token, :success], %{count: count}, %{audience: audience}, _config) do
+      Statix.increment("retrieve_token:success", count, tags: ["audience:#{audience}"])
+    end
+  end
+end
```

Comment:
We can easily detect the Statix module too, for instance taking the first configured module:
```
iex> Application.get_all_env(:statix)
[
  {Vito.Statix, ...}
]
```

Reply:
Personally, I wouldn't add any "complicated" module detection I like the idea of letting the user provide a telemetry reporter so that they can leverage the pre-defined handler! It seems the simplest thing we can offer to avoid having users write a lot of boiler-plate code